File: src/recommendation/logger.py
#!/usr/bin/python

# Copyright The OpenTelemetry Authors
# SPDX-License-Identifier: Apache-2.0
# DataDog Migration - Updated JSON Logger

import logging
import sys
from pythonjsonlogger import jsonlogger
import ddtrace


class CustomJsonFormatter(jsonlogger.JsonFormatter):
    def add_fields(self, log_record, record, message_dict):
        super(CustomJsonFormatter, self).add_fields(log_record, record, message_dict)
        # DataDog trace context injection
        span = ddtrace.tracer.current_span()
        if span:
            log_record['dd.trace_id'] = str(span.trace_id)
            log_record['dd.span_id'] = str(span.span_id)
            log_record['dd.service'] = span.service
        else:
            log_record['dd.trace_id'] = '0'
            log_record['dd.span_id'] = '0'

def getJSONLogger(name):
    logger = logging.getLogger(name)
    handler = logging.StreamHandler(sys.stdout)
    formatter = CustomJsonFormatter('%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] [dd.trace_id=%(dd.trace_id)s dd.span_id=%(dd.span_id)s] - %(message)s')
    handler.setFormatter(formatter)
    logger.addHandler(handler)
    logger.setLevel(logging.INFO)
    logger.propagate = False
    return logger

File: src/payment/logger.js
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0
// DataDog Migration - Payment Service Logger

const pino = require('pino')

const logger = pino({
  mixin() {
    return {
      'service.name': process.env['DD_SERVICE'] || 'payment',
      'service.version': process.env['DD_VERSION'] || '1.0.0',
      'env': process.env['DD_ENV'] || 'development'
    }
  },
  formatters: {
    level: (label) => {
      return { 'level': label };
    },
  },
});

module.exports = logger;
